lib/tasks/generate.rake

```diff
@@ -0,0 +1,85 @@
+require 'json'
+require 'csv'
+require 'util/metadata_tools'
+
+namespace :generate do
+
+  desc "Generate a data.json file"
+  task :datajson, [:filename] => :environment do |_, args|
+
+    # We'll export this in blobs so that we don't load the
+    # entire database into memory.
+    file = File.new("data.json", "w+")
+
+    generate_string().each do | blob |
+      file.puts blob
+    end
+
+    file.close()
+    File.rename "data.json", "public/data.json"
+
+  end
+
+end
+
+# Generates a string to write to the output file
+def generate_string()
```

Comment:
Rather than generate a 90Mb JSON file in memory (along with the associated models and extra data) this method yields data as it is called so that we've only got one map and one json blob in ram at any one time.  May be slightly slower, but it's the closest to a stream of json objects to be written that we can get.